scRNA-seq-advanced/02-celltype_assignment.Rmd

```diff
@@ -0,0 +1,465 @@
+
+We'll do a bit of wrangling to pull out the ADT data, transpose it so each row is a cell,
+then convert it into a data frame for plotting. 
+(We have to first convert to a "regular" R matrix or R won't know how to convert it to a data frame)
+
+```{r}
+# convert to a data frame
+adt_df <- logcounts(altExp(sce)) %>%
+  t() %>% 
+  as.matrix() %>%
+  as.data.frame()
+```
+Now we can make a density plot with `ggplot2`!
+
+```{r}
+ggplot(adt_df, aes(x = CD3)) + 
+  geom_density() +
+  theme_bw()
+```
+
+Looking at a few of these, we can define some rough cutoffs.
+This is very manual and not particularly principled, but it _is_ a potential way to assign cell types.
+
+```{r}
+adt_df <- adt_df %>%
+  dplyr::mutate(
+    celltype = dplyr::case_when(
+      CD3 > 6.7 & CD4 > 8 ~ "CD4+ T cell",
+      CD3 > 6.7 & CD8 > 6 ~ "CD8+ T cell",
+      CD3 > 6.7 ~ "T cell"
+    )
+  )
+```
+
+```{r}
+sce$threshold_celltype <- adt_df$celltype
+scater::plotUMAP(sce, 
+                 colour_by = "threshold_celltype")
+```
+
+We could do similar things for any marker gene from the expression data as well.
+
+A more careful version of this is `AUCell`... do we want to introduce this?
+
+
+## Cell type annotation with `SingleR`
+
+Get some cell data
+```{r}
+# Don't ask about creating the cache directories, just create them
+ExperimentHub::setExperimentHubOption("ASK", FALSE)
+AnnotationHub::setAnnotationHubOption("ASK", FALSE)
+
+# Get Data from the Blueprint & Encode projects from celldex
+blueprint_encode <- celldex::BlueprintEncodeData(ensembl = TRUE)
+```
+What is this `blueprint_encode`?
+
+```{r}
+blueprint_encode
+```
+
+A SummarizedExperiment is very similar to a SingleCellExperiment, except rather than having each column as a cell, each column is a sample.
+Otherwise, the components are very similar: each row is still a gene, for example, and additional data about the samples
+are stored in the `colData`.
+
+What information do we have for the samples?
+
+```{r}
+colData(blueprint_encode)
+```
+
+Let's see how many samples of each type we have:
+
+```{r}
+data.frame(colData(blueprint_encode)) %>% 
+  dplyr::count(label.main, label.fine)
+```
+
+### What does `SingleR` do?
+
+First we build a model that we can use for classifying expression.
+Some options we might explore: how to find choose genes to include in the model.
+`genes = "de"` is the default, using genes differentially expressed among the labels
+`genes = "sd"`uses genes with the most variability across the data set.
+
+This time we will do the training and classification in a single step, using the convenience function `SingleR::SingleR()`.
+
+```{r}
+singler_result <- SingleR::SingleR(
+  sce,
+  ref = blueprint_encode,
+  labels = blueprint_encode$label.main
+)
+```
+
+Plot the results:
+```{r}
+SingleR::plotScoreHeatmap(singler_result)
+```
+
+
+```{r}
+sce$celltype_main <- singler_result$pruned.labels
+```
+
+```{r}
+scater::plotUMAP(sce, colour_by = "celltype_main")
```

Comment:
I just want to note how annoyingly perfect and unrealistic I find this plot. Way to go PBMCs...